src/edu/csus/ecs/pc2/core/execute/IExecuteTimerFrame.java

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 1989-2019 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+/**
+ * IExecuteTimerFrame interface
+ * This interface should be implemented by classes that use the Executable class.
+ * Typically, this is a Frame, such as AutoJudgeStatusFrame or ExecuteTimerFrame
+ */
+package edu.csus.ecs.pc2.core.execute;
+
+import java.awt.Color;
+
+/**
+ * @author John Buck
+ *
+ */
+public interface IExecuteTimerFrame {
+   
+    /**
+     * sets the following frame components to their initial "reset" states
+     * Timer label
+     * Timer counter
+     * Timer terminate button
+     */
+    public void resetFrame();
+    
+    /**
+     * show or hide the timer frame
+     * 
+     * @param bVis
+     */
+    public void setTimerFrameVisible(boolean bVis);
+    
+    /**
+     * change the color of the timer text (for the "Test" button if execute time gets too big)
+     * @param fg
```

Comment:
I do not understand this comment.  It talks about a "Test" button... but as far as I can see there's no such button on the implementing frame.

Reply:
Recall that the same ExecuteTimerFrame is used in the Team model when a team selects the "Test" button.  In this case, the run is allowed to go past the assigned time limit and the CPU time turns red indicating it has exceeded the time limit.

Reply:
Ok.  But doesn't this mean that some implementers of this Interface will have the need to change the Timer label color, while others will not (because they weren't invoked from a frame containing a "Test" button)?   Which in turn means that implementers of the Interface that do NOT need to have a changeable-color capability are _required_ to implement a method which is irrelevant to them.  (It's true that the "implementation" can be trivial -- simply `return` -- but (in IMHO) when an interface requires implementing a method that's not needed, it indicates a potential flaw in the Interface design.)

In particular, it seems like there should be a "super-interface" that provides a default implementation of the method (e.g. one that simply executes `return`), and a "sub-interface" that doesn't mention the method.  That way, classes which don't need the ability to change the label color implement the sub-interface (and inherit the super-interface default implementation) without ever having to pay attention to the notion of changing the color, while classes which do need to change the color can override the method.  This seems like it better follows OO design principles.

It could also be turned around the other way:  have the super-interface _not mention_ the method (and classes which have no need for the color-change capability implement that interface), then have a derived Interface which simply adds the color-change method (and classes which need that method implement the sub-interface).  I'm not sure which of these two approaches is "better"...

And note:  I'm not asking for such a change; I wouldn't consider this item by itself a reason for not approving the PR.  I'm just pointing out what I think is a (slight) flaw in the design.

Reply:
Actually, I take that back - I misread the code (`!isTeam()` instead of `isTeam()`).  In fact, the timer label is only set to red if the client type is NOT team and the time has been exceeded.  I confirmed this operation.  This is not new functionality and `ExecuteTimer `has always set the color.  However, the implementation of how it is done IS new (using the interface).  The implementation of this method is required to provide a visual cue indicating time has been exceeded.  I will update the comment in the interface to reflect its actual functionality in more detail.